diff --git a/node_modules/resolve-url-loader/lib/engine/postcss.js b/node_modules/resolve-url-loader/lib/engine/postcss.js
index 87571f1..9ce66fd 100644
--- a/node_modules/resolve-url-loader/lib/engine/postcss.js
+++ b/node_modules/resolve-url-loader/lib/engine/postcss.js
@@ -30,7 +30,45 @@ function process(sourceFile, sourceContent, params) {
 
   // prepend file protocol to all sources to avoid problems with source map
   return postcss([
-    postcss.plugin('postcss-resolve-url', postcssPlugin)
+    {
+      postcssPlugin: 'postcss-resolve-url',
+      Declaration: function declarationVisitor(declaration) {
+        var prefix,
+            isValid = declaration.value && (declaration.value.indexOf('url') >= 0);
+        if (isValid) {
+          prefix = declaration.prop + declaration.raws.between;
+          declaration.value = params.transformDeclaration(declaration.value, getPathsAtChar);
+        }
+
+        function getPathsAtChar(index) {
+          var subString    = declaration.value.slice(0, index),
+              posSelector  = algerbra.sanitise(declaration.parent.source.start),
+              posProperty  = algerbra.sanitise(declaration.source.start),
+              posValue     = algerbra.add([posProperty, algerbra.strToOffset(prefix)]),
+              posSubString = algerbra.add([posValue, algerbra.strToOffset(subString)]);
+
+          var result = {
+            subString: positionToOriginalDirectory(posSubString),
+            value    : positionToOriginalDirectory(posValue),
+            property : positionToOriginalDirectory(posProperty),
+            selector : positionToOriginalDirectory(posSelector)
+          };
+
+          var isValid = [result.subString, result.value, result.property, result.selector].every(Boolean);
+          if (isValid) {
+            return result;
+          }
+          else if (params.sourceMapConsumer) {
+            throw new Error(
+              'source-map information is not available at url() declaration ' +
+              (ORPHAN_CR_REGEX.test(sourceContent) ? '(found orphan CR, try removeCR option)' : '(no orphan CR found)')
+            );
+          } else {
+            throw new Error('a valid source-map is not present (ensure preceding loaders output a source-map)');
+          }
+        }
+      }
+    }
   ])
     .process(correctedContent, {
       from: fileProtocol.prepend(sourceFile),
@@ -46,65 +84,6 @@ function process(sourceFile, sourceContent, params) {
       map    : params.outputSourceMap ? fileProtocol.remove(result.map.toJSON()) : null
     }));
 
-  /**
-   * Plugin for postcss that follows SASS transpilation.
-   */
-  function postcssPlugin() {
-    return function applyPlugin(styles) {
-      styles.walkDecls(eachDeclaration);
-    };
-
-    /**
-     * Process a declaration from the syntax tree.
-     * @param declaration
-     */
-    function eachDeclaration(declaration) {
-      var prefix,
-          isValid = declaration.value && (declaration.value.indexOf('url') >= 0);
-      if (isValid) {
-        prefix = declaration.prop + declaration.raws.between;
-        declaration.value = params.transformDeclaration(declaration.value, getPathsAtChar);
-      }
-
-      /**
-       * Create a hash of base path strings.
-       *
-       * Position in the declaration is supported by postcss at the position of the url() statement.
-       *
-       * @param {number} index Index in the declaration value at which to evaluate
-       * @throws Error on invalid source map
-       * @returns {{subString:string, value:string, property:string, selector:string}} Hash of base path strings
-       */
-      function getPathsAtChar(index) {
-        var subString    = declaration.value.slice(0, index),
-            posSelector  = algerbra.sanitise(declaration.parent.source.start),
-            posProperty  = algerbra.sanitise(declaration.source.start),
-            posValue     = algerbra.add([posProperty, algerbra.strToOffset(prefix)]),
-            posSubString = algerbra.add([posValue, algerbra.strToOffset(subString)]);
-
-        var result = {
-          subString: positionToOriginalDirectory(posSubString),
-          value    : positionToOriginalDirectory(posValue),
-          property : positionToOriginalDirectory(posProperty),
-          selector : positionToOriginalDirectory(posSelector)
-        };
-
-        var isValid = [result.subString, result.value, result.property, result.selector].every(Boolean);
-        if (isValid) {
-          return result;
-        }
-        else if (params.sourceMapConsumer) {
-          throw new Error(
-            'source-map information is not available at url() declaration ' +
-            (ORPHAN_CR_REGEX.test(sourceContent) ? '(found orphan CR, try removeCR option)' : '(no orphan CR found)')
-          );
-        } else {
-          throw new Error('a valid source-map is not present (ensure preceding loaders output a source-map)');
-        }
-      }
-    }
-  }
-
   /**
    * Given an apparent position find the directory of the original file.
    *
